aten/src/ATen/native/cuda/ForeachPointwiseOp.cu

```diff
@@ -160,106 +182,165 @@ std::vector<Tensor> foreach_pointwise_op(
-      TensorList tensors1,                                                \
-      TensorList tensors2,                                                \
-      const Tensor& scalars_) {                                           \
-    auto scalars = convert_tensor_to_scalar_list(scalars_, input.size()); \
-    check_foreach_api_restrictions(input, tensors1, tensors2, scalars);   \
-    if (!can_use_fast_route({input, tensors1, tensors2}) ||               \
-        has_integral_tensor(input, /* includeBool */ true)) {             \
-      return at::native::foreach_tensor_##NAME##_scalarlist_slow(         \
-          input, tensors1, tensors2, scalars);                            \
-    }                                                                     \
-                                                                          \
-    return foreach_pointwise_op<OP>(input, tensors1, tensors2, scalars);  \
-  }                                                                       \
-                                                                          \
-  void foreach_tensor_##NAME##_tensor_cuda_(                              \
-      TensorList input,                                                   \
-      TensorList tensors1,                                                \
-      TensorList tensors2,                                                \
-      const Tensor& scalars_) {                                           \
-    auto scalars = convert_tensor_to_scalar_list(scalars_, input.size()); \
-    check_foreach_api_restrictions(input, tensors1, tensors2, scalars);   \
-    if (!can_use_fast_route({input, tensors1, tensors2}, scalars) ||      \
-        has_integral_tensor(input, /* includeBool */ true)) {             \
-      return at::native::foreach_tensor_##NAME##_scalarlist_slow_(        \
-          input, tensors1, tensors2, scalars);                            \
-    }                                                                     \
-                                                                          \
-    foreach_pointwise_op_<OP>(input, tensors1, tensors2, scalars);        \
+#define FOREACH_POINTWISE_OP_TENSOR(NAME, OP)                               \
+  std::vector<Tensor> foreach_tensor_##NAME##_tensor_cuda(                  \
+      TensorList input,                                                     \
+      TensorList tensors1,                                                  \
+      TensorList tensors2,                                                  \
+      const Tensor& scalars_) {                                             \
+    auto scalars = convert_tensor_to_scalar_list(scalars_, input.size());   \
+    check_foreach_api_restrictions(input, tensors1, tensors2, scalars);     \
+    std::pair<bool, bool> p =                                               \
+        can_use_fast_route({input, tensors1, tensors2}, scalars);           \
```

Comment:
Note that I've added scalars to the can_use_fast_route call here. I'm pretty sure this should have been added before but want to confirm with @crcrpar